src/mapgen/mapgen.cpp

```diff
@@ -697,6 +698,11 @@ void MapgenBasic::generateBiomes()
 				if (biomemap[index] == BIOME_NONE && is_stone_surface)
 					biomemap[index] = biome->index;
 
+				// Store biome of first water surface detected, as a fallback
+				// entry for the biomemap.
+				if (water_biome_index == 0 && is_water_surface)
+					water_biome_index = biome->index;
```

Comment:
`->index` is `u32` but your variable has the type `u8`

Reply:
Hmm, i did this because the biomemap uses u8 vales, however either seems ok, will change it.

Reply:
What you meant is the value of the biome array. However, here it's an index for the array which is another number type; independent.

Reply:
Yes i agree, thanks. Done.